xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/Utils.java

```diff
@@ -365,6 +365,12 @@ public static URI createURI(URI baseURI, java.lang.Class< ? > resourceClass, jav
                             throw new RuntimeException("The 'spaceName' parameter must be a list!");
                         }
                         encodedPathElements[i] = generateEncodedSpacesURISegment((List) pathElements[i]);
+                    // see generateEncodedJobIdURISegment to understand why we manually handle "jobId"
+                    } else if (i < pathVariableNames.size() && "jobId".equals(pathVariableNames.get(i))) {
```

Comment:
I really don't like this. We should not know about "jobId" parameter in this generic location.

Reply:
What I could do to make it more generic is to trigger the condition on "instanceof List", but I'm a bit afraid on the possible side effects on the future. 

Would be better if JAX-RS handled it directly: when I try to pass in argument `id.toArray()` the `encodedPathElements` were properly created but then in `uriBuilder.buildFromEncoded(encodedPathElements)` (https://github.com/xwiki/xwiki-platform/pull/1296/files#diff-3d532b9e95503f09519b75a7525fc3beR388) only the first element was used. Maybe you have an idea to tell it to use all elements?

Reply:
If what you mean is that passing several elements to Utils#createURI does not work then that's a huge bug and it needs to be fixed since it means the contract is not respected and it does not make any sense to take a `java.lang.Object... pathElements`.